tchannel/tornado/connection.py

```diff
@@ -192,6 +195,8 @@ def on_error(future):
 
             if isinstance(exception, tornado.iostream.StreamClosedError):
                 self.close()
+            if message_future.running():
```

Comment:
Lots of our tests actually cover this case. For example a simple client & server case, client sends one request to server, gets response and client exists. At the end, this connection is closed however the server side's , it ``yield _recv()`` and this message_future never got triggered.